src/hooks/useTokenPrice.js

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from 'react'
+import { getMarketDetails, getTokenReserves } from '@1hive/uniswap-sdk'
+import { useWallet } from '../providers/Wallet'
+import BigNumber from '../lib/bigNumber'
+import { formatTokenAmount } from '../lib/token-utils'
+
+const UNISWAP_PRECISION = 18
+const UNISWAP_MARKET_RETRY_EVERY = 1000
+
+async function getTokenMarketDetails(tokenAddress, provider) {
+  const tokenReserves = await getTokenReserves(tokenAddress, provider)
+
+  // For XDAI we must pass undefined
+  return getMarketDetails(undefined, tokenReserves)
```

Comment:
Is not easy way to do the changes on the uniswap lib and pass another thing rather than undefined?

Reply:
Note that that's just how the sdk works. https://uniswap.org/docs/v1/SDK/computation/

For `ETH` you have to pass `undefined` as the token reserves.